packages/arb-ts/token_lists/lists/token-list-42161.json

```diff
@@ -0,0 +1,103 @@
+{
+  "name": "Arb1",
+  "timestamp": "1623877437368",
+  "version": { "major": 0, "minor": 1, "patch": 0 },
+  "tokens": [
+    {
+      "chainId": 42161,
+      "address": "0xFa7F8980b0f1E64A2062791cc3b0871572f1F7f0",
+      "name": "Uniswap",
+      "symbol": "UNI",
+      "decimals": 18,
+      "logoURI": "ipfs://QmXttGpZrECX5qCyXbBQiqgQNytVGeZW5Anewvh2jc4psg",
+      "extensions": {
+        "l1Address": "0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984",
+        "l2TokenType": "standard"
+      }
+    },
+    {
+      "chainId": 42161,
+      "address": "0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8",
+      "name": "USDCoin",
+      "symbol": "USDC",
+      "decimals": 6,
+      "logoURI": "https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/logo.png",
+      "extensions": {
+        "l1Address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+        "l2TokenType": "arbCustom"
+      }
+    },
+    {
+      "chainId": 42161,
+      "address": "0x2e9a6Df78E42a30712c10a9Dc4b1C8656f8F2879",
+      "name": "Maker",
+      "symbol": "MKR",
+      "decimals": 18,
+      "logoURI": "https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2/logo.png",
+      "extensions": {
+        "l1Address": "0x9f8F72aA9304c8B593d555F12eF6589cC3A579A2",
+        "l2TokenType": "standard"
+      }
+    },
+    {
+      "chainId": 42161,
+      "address": "0xf97f4df75117a78c1A5a0DBb814Af92458539FB4",
+      "name": "ChainLink Token",
+      "symbol": "LINK",
+      "decimals": 18,
+      "logoURI": "https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0x514910771AF9Ca656af840dff83E8264EcF986CA/logo.png",
+      "extensions": {
+        "l1Address": "0x514910771AF9Ca656af840dff83E8264EcF986CA",
+        "l2TokenType": "standard"
+      }
+    },
+    {
+      "chainId": 42161,
+      "address": "0x2f2a2543B76A4166549F7aaB2e75Bef0aefC5B0f",
+      "name": "Wrapped BTC",
+      "symbol": "WBTC",
+      "decimals": 8,
+      "logoURI": "https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599/logo.png",
+      "extensions": {
+        "l1Address": "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",
+        "l2TokenType": "standard"
+      }
+    },
+    {
+      "chainId": 42161,
+      "address": "0x23A941036Ae778Ac51Ab04CEa08Ed6e2FE103614",
+      "name": "The Graph",
+      "symbol": "GRT",
+      "decimals": 18,
+      "logoURI": "https://assets.coingecko.com/coins/images/13397/thumb/Graph_Token.png?1608145566",
+      "extensions": {
+        "l1Address": "0xc944E90C64B2c07662A292be6244BDf05Cda44a7",
+        "l2TokenType": "standard"
```

Comment:
Instead of token type, we could specify the L2 gateway address and derive if its standard/custom/etc from that

Reply:
this is effectively doing that already? (just via the L1Gateway address, in the conditional)

Reply:
i guess maybe gatewayAddress as a field could be nicer

Reply:
updated to use L1gateway address